tests/subsys/net/lib/nrf_cloud/fota_common/src/main.c

```diff
@@ -0,0 +1,823 @@
+
+	ret = nrf_cloud_pending_fota_job_process(NULL, &reboot_required);
+
+	zassert_equal(-EINVAL, ret,
+		      "return should be -EINVAL when job is NULL");
+}
+
+/* Verify that nrf_cloud_pending_fota_job_process fails when the job parameters are invalid. */
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_pending_fota_job_process_job_not_applicable)
+{
+	struct nrf_cloud_settings_fota_job job;
+	bool reboot_required;
+
+	job.validate = NRF_CLOUD_FOTA_VALIDATE_NONE;
+	job.type = NRF_CLOUD_FOTA_MODEM_DELTA;
+
+	int ret = nrf_cloud_pending_fota_job_process(&job, &reboot_required);
+
+	zassert_equal(-ENODEV, ret,
+		      "return should be -ENODEV when job is not NRF_CLOUD_FOTA_VALIDATE_PENDING");
+
+	job.validate = NRF_CLOUD_FOTA_VALIDATE_PENDING;
+	job.type = NRF_CLOUD_FOTA_TYPE__INVALID;
+
+	ret = nrf_cloud_pending_fota_job_process(&job, &reboot_required);
+
+	zassert_equal(-ENODEV, ret,
+		      "return should be -ENODEV when job type is NRF_CLOUD_FOTA_TYPE__INVALID");
+}
+
+/* Verify that nrf_cloud_pending_fota_job_process succeeds when a pending
+ * APP FOTA validation PASSES.
+ */
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_pending_fota_job_process_app_pass)
+{
+	Z_TEST_SKIP_IFNDEF(CONFIG_MCUBOOT_IMG_MANAGER);
+	struct nrf_cloud_settings_fota_job job;
+	bool reboot_required;
+
+	job.validate = NRF_CLOUD_FOTA_VALIDATE_PENDING;
+	job.type = NRF_CLOUD_FOTA_APPLICATION;
+
+	boot_is_img_confirmed_fake.custom_fake = fake_boot_is_img_confirmed__false;
+	boot_write_img_confirmed_fake.custom_fake = fake_boot_write_img_confirmed__succeeds;
+
+	int ret = nrf_cloud_pending_fota_job_process(&job, &reboot_required);
+
+	zassert_equal(0, ret, "return should be 0 when app FOTA succeeds");
+	zassert_equal(NRF_CLOUD_FOTA_VALIDATE_PASS, job.validate,
+		      "validate status should be PASS on success");
+}
+
+/* Verify that nrf_cloud_pending_fota_job_process succeeds when a pending
+ * APP FOTA validation FAILS.
+ */
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_pending_fota_job_process_app_fail)
+{
+	Z_TEST_SKIP_IFNDEF(CONFIG_MCUBOOT_IMG_MANAGER);
+	struct nrf_cloud_settings_fota_job job;
+	bool reboot_required = false;
+
+	job.validate = NRF_CLOUD_FOTA_VALIDATE_PENDING;
+	job.type = NRF_CLOUD_FOTA_APPLICATION;
+
+	boot_is_img_confirmed_fake.custom_fake = fake_boot_is_img_confirmed__false;
+	boot_write_img_confirmed_fake.custom_fake = fake_boot_write_img_confirmed__fails;
+
+	int ret = nrf_cloud_pending_fota_job_process(&job, &reboot_required);
+
+	zassert_equal(0, ret, "return should be 0 when app FOTA is processed as failure");
+	zassert_equal(true, reboot_required,
+		      "reboot_required should be true when app FOTA is processed as failure");
+	zassert_equal(NRF_CLOUD_FOTA_VALIDATE_FAIL, job.validate,
+		      "validate status should be FAIL when app FOTA is processed as failure");
+}
+
+/* Verify that nrf_cloud_pending_fota_job_process succeeds when a pending
+ * APP FOTA validation is UNKNOWN.
+ */
```

Comment:
I want to rephrase this as `Verify that nrf_cloud_pending_fota_job_process succeeds and sets validate status to UNKNOWN if XXX`

But the intended condition XXX is unclear to me

Is UNKNOWN just always expected for APP FOTA jobs? How does `boot_is_img_confirmed` relate to it?